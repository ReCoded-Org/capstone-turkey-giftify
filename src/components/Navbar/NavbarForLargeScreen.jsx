import React from "react";
import { Link } from "react-router-dom";
import { Button } from "react-bootstrap";
import { useSelector, useDispatch } from "react-redux";
import firebase from "firebase/app";
import "firebase/auth";
import { largeScreenNavbarData } from "../../data/Navbar/largeScreenNavbarData.js";
import navBarLogo from "../../images/navBarLogo.png";

const NavbarForLargeScreen = () => {
  const { userId, userName } = useSelector((state) => state.user);
  const displayProfile = userId === "" ? "none" : "inline-block";
  const displayLogin = userId === "" ? "inline-block" : "none";
  const dispatch = useDispatch();

  const LogOut = () => {
    firebase
      .auth()
      .signOut()
      .then(() => {
        dispatch({
          type: "user/logout",
        });
        // Sign-out successful.
      })
      .catch((error) => {
        alert(error);
        // An error happened.
      });
  };

  return (
    <div className="navbar">
      <div className="navbarLogo">
        <Link to="/">
          <Image src={navBarLogo} />
        </Link>
      </div>
      <div className="navbarLinks">
        {largeScreenNavbarData.map((info) => (
          <Link to={info.target} className="eachLinksInNavbar" key={info.id}>
            {info.name}
          </Link>
        ))}
      </div>
      <div className="navbarProfile">
        <Link to="/logIn">
          <Button
            variant="primary"
            style={{
              borderRadius: "0.75rem",
              marginBottom: "5%",
              display: displayLogin,
            }}
          >
            Login
          </Button>
        </Link>
        <Link to="/signUp">
          <Button
            variant="primary"
            style={{
              borderRadius: "0.75rem",
              marginBottom: "5%",
              display: displayLogin,
            }}
          >
            Sign Up
          </Button>
        </Link>
        <Link to="/userProfile">
          <Button
            variant="primary"
            style={{
              borderRadius: "0.75rem",
              marginBottom: "5%",
              display: displayProfile,
            }}
          >
            {userName}
          </Button>
        </Link>
        <Link to="/">
          <Button
            variant="primary"
            onClick={LogOut}
            style={{
              borderRadius: "0.75rem",
              marginBottom: "5%",
              display: displayProfile,
            }}
          >
            Log Out
          </Button>
        </Link>
      </div>
    </div>
  );
};

export default NavbarForLargeScreen;
