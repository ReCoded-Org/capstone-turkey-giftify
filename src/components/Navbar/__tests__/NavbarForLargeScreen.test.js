import React from "react";
import NavbarForLargeScreen from "./../NavbarForLargeScreen";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";

afterEach(cleanup);

it("NavbarForLargeScreen rendered correctly", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <NavbarForLargeScreen />
      </BrowserRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
