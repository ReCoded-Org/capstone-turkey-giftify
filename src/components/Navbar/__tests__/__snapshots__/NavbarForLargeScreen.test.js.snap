// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavbarForLargeScreen rendered correctly 1`] = `
<div
  className="navbar"
>
  <div
    className="navbarLogo"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        className=""
        src="navBarLogo.png"
      />
    </a>
  </div>
  <div
    className="navbarLinks"
  >
    <a
      className="eachLinksInNavbar"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="eachLinksInNavbar"
      href="/search"
      onClick={[Function]}
    >
      Search
    </a>
    <a
      className="eachLinksInNavbar"
      href="/requests"
      onClick={[Function]}
    >
      Requests
    </a>
    <a
      className="eachLinksInNavbar"
      href="/donations"
      onClick={[Function]}
    >
      Donations
    </a>
    <a
      className="eachLinksInNavbar"
      href="/about"
      onClick={[Function]}
    >
      About
    </a>
    <a
      className="eachLinksInNavbar"
      href="/contactUs"
      onClick={[Function]}
    >
      Contact Us
    </a>
    <div
      className="dropdownBasicButton dropdown"
    >
      <button
        aria-expanded={false}
        aria-haspopup={true}
        className="dropdown-toggle btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Language
      </button>
    </div>
  </div>
  <div
    className="navbarProfile"
  >
    <a
      href="/logIn"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        disabled={false}
        style={
          Object {
            "borderRadius": "0.75rem",
            "display": "inline-block",
            "marginBottom": "5%",
          }
        }
        type="button"
      >
        Login
      </button>
    </a>
    <a
      href="/signUp"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        disabled={false}
        style={
          Object {
            "borderRadius": "0.75rem",
            "display": "inline-block",
            "marginBottom": "5%",
          }
        }
        type="button"
      >
        Sign Up
      </button>
    </a>
    <a
      href="/userProfile"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        disabled={false}
        style={
          Object {
            "borderRadius": "0.75rem",
            "display": "none",
            "marginBottom": "5%",
          }
        }
        type="button"
      >
        
      </button>
    </a>
    <a
      href="/"
      onClick={[Function]}
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        style={
          Object {
            "borderRadius": "0.75rem",
            "display": "none",
            "marginBottom": "5%",
          }
        }
        type="button"
      >
        Log Out
      </button>
    </a>
  </div>
</div>
`;
